zio-schema/shared/src/main/scala/zio/schema/Schema.scala

```diff
@@ -816,11 +816,14 @@ object Schema {
         }
       )
 
-  sealed trait Record[A] extends Schema[A] {
-    def structure: ListMap[String, Schema[_]]
+  final case class Field[A](label: String, schema: Schema[A], annotations: Seq[Any] = Nil)
+
+  sealed trait Record[R] extends Schema[R] {
+    def structure: Seq[Field[_]]
+    def annotations: Seq[Any] = Nil
   }
 
-  final case class GenericRecord(override val structure: ListMap[String, Schema[_]]) extends Record[ListMap[String, _]]
+  final case class GenericRecord(override val structure: Seq[Field[_]]) extends Record[ListMap[String, _]]
```

Comment:
It's ok for `Seq` to appear in contravariant position (as an input to a method), but it should not appear in covariant position (as an output from a method) because it provides no _Big Oh_ guarantees on any of its methods.

Instead, I suggest we refactor this to be `Chunk[Field[_]]` here and elsewhere, so that users have guaranteed `O(1)` random access, etc.

Reply:
Makes sense